labcontrol/db/support_files/populate_test_db.sql

```diff
@@ -877,6 +877,15 @@ BEGIN
     ------------------------------------------
     ------ SEQUENCING PROCESS ERROR CASE------
     ------------------------------------------
+    /* This error case appears to be a copy of Shotgun entry above, with a
+       different value for the assay name. The assay column has been removed
+       from the sequencing_process table, and the relevant code now queries
+       PoolComposition to return the correct value for assay_type based on
+       constituent composite types (Something previously done in
+       SequencingProcess.create() code.
+     
+       Hence, this error case will no longer produce the desired result in
+       tests, and no longer a valid source of error.*/
```

Comment:
That is correct; unittests can be constructed such that the job of the assert is to intentionally catch a class of Error, and it will raise an Error of its own or otherwise report back through the unittest framework if an Error wasn't raised.

Previously there was a duplicate of the INSERT statements used to directly create the shotgun metadata in the database; this test used otherwise valid data, but changed the value of the assay_type to a non 'Amplicon' or 'Metagenomic' value. This was done as a sanity check because assay_types were just hardcoded strings, and not any sort of enumerated type.

Now that the types are enumerated in the database, the third test that was previously in this location no longer makes sense to have, and was therefore removed.

Reply:
Also, as this is a public project, it might good to refrain from documenting your level of SQL knowledge here.